app/javascript/src/NewService/components/ServiceDiscoveryForm.jsx

```diff
@@ -0,0 +1,40 @@
+// @flow
+
+import React from 'react'
+import {CSRFToken} from 'utilities/utils'
+import type {ServiceDiscoveryFormType} from 'NewService/types'
+
+const renderSelectOptions = (options: string[]) => {
+  if (options.length > 0) {
+    return options.map(option => <option key={option} value={option}>{option}</option>)
+  }
+}
+
+const ServiceDiscoveryForm = (props: ServiceDiscoveryFormType) => {
+  const {isVisible, providerAdminServiceDiscoveryServicesPath, namespaces, services, selectedNamespace, handleChangeNamespaces} = props
+  return (<form className={`formtastic ${isVisible ? '' : 'is-hidden'}`} id="service_discovery" action={providerAdminServiceDiscoveryServicesPath} acceptCharset="UTF-8" method="post">
```

Comment:
You are using the same form here and in `ServiceNewForm`, you might want to extract that one in a separated component